sim-lib/src/lib.rs

```diff
@@ -538,32 +555,58 @@ impl Simulation {
         tasks: &mut JoinSet<()>,
     ) {
         let listener = self.shutdown_listener.clone();
+        let shutdown = self.shutdown_trigger.clone();
         log::debug!("Setting up simulator data collection.");
 
         // Create a sender/receiver pair that will be used to report final results of simulation.
         let (results_sender, results_receiver) = channel(1);
 
-        tasks.spawn(produce_simulation_results(
-            self.nodes.clone(),
-            output_receiver,
-            results_sender,
-            listener.clone(),
-        ));
+        let nodes = self.nodes.clone();
+        // psr: produce simulation results
+        let psr_listener = listener.clone();
+        let psr_shutdown = shutdown.clone();
+        tasks.spawn(async move {
+            log::debug!("Starting simulation results producer.");
+            if let Err(e) =
+                produce_simulation_results(nodes, output_receiver, results_sender, psr_listener)
+                    .await
+            {
+                psr_shutdown.trigger();
+                log::error!("Produce simulation results exited with error: {e:?}.");
+            }
+        });
 
         let result_logger = Arc::new(Mutex::new(PaymentResultLogger::new()));
 
-        tasks.spawn(run_results_logger(
-            listener.clone(),
-            result_logger.clone(),
-            Duration::from_secs(60),
-        ));
+        let result_logger_clone = result_logger.clone();
+        let result_logger_listener = listener.clone();
+        tasks.spawn(async move {
+            log::debug!("Starting results logger.");
+            run_results_logger(
+                result_logger_listener,
+                result_logger_clone,
+                Duration::from_secs(60),
+            )
+            .await
```

Comment:
For consistency, let's log exit here like we do for the other spawned tasks. 

Also wasn't initially clear why we're not also triggering here. Could you add a comment that `run_results_logger` only exits on `listener` trigger so we don't need to handle erroring out?